client/hotelDetails.jsx

```diff
@@ -0,0 +1,110 @@
+
+import React from 'react';
+
+class HotelDetails extends React.Component {
+
+  render() {
+    const hotelList = this.props.data.hotelList;
+    let selectedIndex = null;
+    for (let i = 0; i < hotelList.length; i++) {
+      if (hotelList[i].name === this.props.data.selectedHotel) {
+        selectedIndex = i;
+        break;
+      }
+    }
```

Comment:
Instead of forcing this `HotelDetails` component to find its own data, move this logic into the parent component.